<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analyzing circadian transcriptome data with LimoRhyde • limorhyde</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analyzing circadian transcriptome data with LimoRhyde">
<meta property="og:description" content="limorhyde">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">limorhyde</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/introduction.html">Analyzing circadian transcriptome data with LimoRhyde</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/hugheylab/limorhyde/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="introduction_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analyzing circadian transcriptome data with LimoRhyde</h1>
            
            <h4 class="date">2020-08-11</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/hugheylab/limorhyde/blob/master/vignettes/introduction.Rmd"><code>vignettes/introduction.Rmd</code></a></small>
      <div class="hidden name"><code>introduction.Rmd</code></div>

    </div>

    
    
<p>LimoRhyde is a framework for differential analysis of rhythmic transcriptome data. This vignette goes through the typical steps of an analysis: identifying rhythmic genes, identifying differentially rhythmic genes, and identifying differentially expressed genes. The dataset is based on total RNA from livers of wild-type and Rev-erb<span class="math inline">\(\alpha/\beta\)</span> knockout mice, with gene expression measured by microarray (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE34018">GSE34018</a>).</p>
<div id="load-packages-and-set-parameters" class="section level2">
<h2 class="hasAnchor">
<a href="#load-packages-and-set-parameters" class="anchor"></a>Load packages and set parameters</h2>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'annotate'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'data.table'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'foreach'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'GEOquery'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'ggplot2'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'knitr'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'limma'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'limorhyde'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">'org.Mm.eg.db'</span>)

<span class="fu"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">'extdata'</span>, <span class="st">'vignette_functions.R'</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">'limorhyde'</span>))</pre></body></html></div>
<p>Here we specify the zeitgeber period and the q-value cutoffs for rhythmic and differentially rhythmic genes.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="no">period</span> <span class="kw">=</span> <span class="fl">24</span>
<span class="no">qvalRhyCutoff</span> <span class="kw">=</span> <span class="fl">0.15</span>
<span class="no">qvalDrCutoff</span> <span class="kw">=</span> <span class="fl">0.1</span></pre></body></html></div>
</div>
<div id="load-the-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#load-the-dataset" class="anchor"></a>Load the dataset</h2>
<p>For simplicity, we use the GEOquery package to load the processed data previously downloaded from NCBI GEO.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">eset</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GEOquery/man/getGEO.html">getGEO</a></span>(<span class="kw">filename</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">'extdata'</span>, <span class="st">'GSE34018_series_matrix.txt'</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">'limorhyde'</span>))</pre></body></html></div>
<p>Now we construct the data.table of sample metadata.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">sm</span> <span class="kw">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html">data.table</a></span>(<span class="fu">pData</span>(<span class="fu">phenoData</span>(<span class="no">eset</span>)))
<span class="no">sm</span> <span class="kw">=</span> <span class="no">sm</span>[, <span class="fu">.</span>(<span class="no">title</span>, <span class="kw">sample</span> <span class="kw">=</span> <span class="no">geo_accession</span>, <span class="kw">genotype</span> <span class="kw">=</span> <span class="no">`genotype/variation:ch1`</span>)]
<span class="no">sm</span>[, <span class="no">time</span> <span class="kw">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span>(<span class="st">'_.*'</span>, <span class="st">''</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span>(<span class="st">'.*_ZT'</span>, <span class="st">''</span>, <span class="no">title</span>)))]
<span class="no">sm</span>[, <span class="no">cond</span> <span class="kw">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="no">genotype</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'wild-type'</span>, <span class="st">'Reverb alpha/beta DKO'</span>),
                    <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'wild-type'</span>, <span class="st">'knockout'</span>))]
<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html">setorderv</a></span>(<span class="no">sm</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'cond'</span>, <span class="st">'time'</span>))

<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">sm</span>[<span class="fl">1</span>:<span class="fl">5</span>, ])</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left">title</th>
<th align="left">sample</th>
<th align="left">genotype</th>
<th align="right">time</th>
<th align="left">cond</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">wild-type liver_ZT0_rep 1</td>
<td align="left">GSM840516</td>
<td align="left">wild-type</td>
<td align="right">0</td>
<td align="left">wild-type</td>
</tr>
<tr class="even">
<td align="left">wild-type liver_ZT0_rep 2</td>
<td align="left">GSM840517</td>
<td align="left">wild-type</td>
<td align="right">0</td>
<td align="left">wild-type</td>
</tr>
<tr class="odd">
<td align="left">wild-type liver_ZT4_rep 1</td>
<td align="left">GSM840518</td>
<td align="left">wild-type</td>
<td align="right">4</td>
<td align="left">wild-type</td>
</tr>
<tr class="even">
<td align="left">wild-type liver_ZT4_rep 2</td>
<td align="left">GSM840519</td>
<td align="left">wild-type</td>
<td align="right">4</td>
<td align="left">wild-type</td>
</tr>
<tr class="odd">
<td align="left">wild-type liver_ZT8_rep 1</td>
<td align="left">GSM840520</td>
<td align="left">wild-type</td>
<td align="right">8</td>
<td align="left">wild-type</td>
</tr>
</tbody>
</table>
<p>Next we use <code>limorhyde</code> to calculate <code>time_cos</code> and <code>time_sin</code>, which are based on the first harmonic of a Fourier decomposition of the <code>time</code> column, and append them to the <code>sm</code> data.table.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">sm</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/cbind.html">cbind</a></span>(<span class="no">sm</span>, <span class="fu"><a href="../reference/limorhyde.html">limorhyde</a></span>(<span class="no">sm</span>$<span class="no">time</span>, <span class="st">'time_'</span>))</pre></body></html></div>
<p>Finally, we calculate the log-transformed expression values in terms of Entrez Gene IDs.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="no">mapping</span> <span class="kw">=</span> <span class="fu">getGeneMapping</span>(<span class="fu">featureData</span>(<span class="no">eset</span>))
<span class="no">emat</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span>(<span class="fu">calcExprsByGene</span>(<span class="no">eset</span>, <span class="no">mapping</span>))[, <span class="no">sm</span>$<span class="no">sample</span>]</pre></body></html></div>
</div>
<div id="identify-rhythmic-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-rhythmic-genes" class="anchor"></a>Identify rhythmic genes</h2>
<p>The next three steps use <a href="https://doi.org/doi:10.18129/B9.bioc.limma">limma</a>. We calculate the q-value of rhythmicity for each gene using that gene’s p-value for each condition and adjusting for multiple testing.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">rhyLimma</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/foreach/man/foreach.html">foreach</a></span>(<span class="kw">condNow</span> <span class="kw">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html">unique</a></span>(<span class="no">sm</span>$<span class="no">cond</span>), <span class="kw">.combine</span> <span class="kw">=</span> <span class="no">rbind</span>) <span class="kw">%do%</span> {
  <span class="no">design</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span>(~ <span class="no">time_cos</span> + <span class="no">time_sin</span>, <span class="kw">data</span> <span class="kw">=</span> <span class="no">sm</span>[<span class="no">cond</span> <span class="kw">==</span> <span class="no">condNow</span>])
  <span class="no">fit</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span>(<span class="no">emat</span>[, <span class="no">sm</span>$<span class="no">cond</span> <span class="kw">==</span> <span class="no">condNow</span>], <span class="no">design</span>)
  <span class="no">fit</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span>(<span class="no">fit</span>, <span class="kw">trend</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
  <span class="no">rhyNow</span> <span class="kw">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html">data.table</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span>(<span class="no">fit</span>, <span class="kw">coef</span> <span class="kw">=</span> <span class="fl">2</span>:<span class="fl">3</span>, <span class="kw">number</span> <span class="kw">=</span> <span class="fl">Inf</span>), <span class="kw">keep.rownames</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
  <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setattr.html">setnames</a></span>(<span class="no">rhyNow</span>, <span class="st">'rn'</span>, <span class="st">'geneId'</span>)
  <span class="no">rhyNow</span>[, <span class="no">cond</span> <span class="kw">:=</span> <span class="no">condNow</span>]
}

<span class="no">rhyLimmaSummary</span> <span class="kw">=</span> <span class="no">rhyLimma</span>[, <span class="fu">.</span>(<span class="kw">P.Value</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span>(<span class="no">P.Value</span>)), <span class="kw">by</span> <span class="kw">=</span> <span class="no">geneId</span>]
<span class="no">rhyLimmaSummary</span>[, <span class="no">adj.P.Val</span> <span class="kw">:=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span>(<span class="no">P.Value</span>, <span class="kw">method</span> <span class="kw">=</span> <span class="st">'BH'</span>)]
<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html">setorderv</a></span>(<span class="no">rhyLimmaSummary</span>, <span class="st">'adj.P.Val'</span>)

<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">rhyLimmaSummary</span>[<span class="fl">1</span>:<span class="fl">5</span>, ])</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left">geneId</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">13170</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="even">
<td align="left">353187</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="odd">
<td align="left">15496</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="even">
<td align="left">12700</td>
<td align="right">0</td>
<td align="right">1e-07</td>
</tr>
<tr class="odd">
<td align="left">321018</td>
<td align="right">0</td>
<td align="right">1e-07</td>
</tr>
</tbody>
</table>
</div>
<div id="identify-differentially-rhythmic-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-differentially-rhythmic-genes" class="anchor"></a>Identify differentially rhythmic genes</h2>
<p>Differential rhythmicity is based on statistical interactions between <code>cond</code> and the <code>time</code> components. We pass all genes to limma (whose Empirical Bayes does best with many genes), but keep results only for rhythmic genes, and adjust for multiple testing accordingly.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="no">design</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span>(~ <span class="no">cond</span> * (<span class="no">time_cos</span> + <span class="no">time_sin</span>), <span class="kw">data</span> <span class="kw">=</span> <span class="no">sm</span>)

<span class="no">fit</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span>(<span class="no">emat</span>, <span class="no">design</span>)
<span class="no">fit</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span>(<span class="no">fit</span>, <span class="kw">trend</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="no">drLimma</span> <span class="kw">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html">data.table</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span>(<span class="no">fit</span>, <span class="kw">coef</span> <span class="kw">=</span> <span class="fl">5</span>:<span class="fl">6</span>, <span class="kw">number</span> <span class="kw">=</span> <span class="fl">Inf</span>), <span class="kw">keep.rownames</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setattr.html">setnames</a></span>(<span class="no">drLimma</span>, <span class="st">'rn'</span>, <span class="st">'geneId'</span>)

<span class="no">drLimma</span> <span class="kw">=</span> <span class="no">drLimma</span>[<span class="no">geneId</span> <span class="kw">%in%</span> <span class="no">rhyLimmaSummary</span>[<span class="no">adj.P.Val</span> <span class="kw">&lt;=</span> <span class="no">qvalRhyCutoff</span>]$<span class="no">geneId</span>]
<span class="no">drLimma</span>[, <span class="no">adj.P.Val</span> <span class="kw">:=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span>(<span class="no">P.Value</span>, <span class="kw">method</span> <span class="kw">=</span> <span class="st">'BH'</span>)]
<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html">setorderv</a></span>(<span class="no">drLimma</span>, <span class="st">'adj.P.Val'</span>)

<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">drLimma</span>[<span class="fl">1</span>:<span class="fl">5</span>, ])</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left">geneId</th>
<th align="right">condknockout.time_cos</th>
<th align="right">condknockout.time_sin</th>
<th align="right">AveExpr</th>
<th align="right">F</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">12686</td>
<td align="right">-1.5326997</td>
<td align="right">-0.0433879</td>
<td align="right">10.157657</td>
<td align="right">63.70799</td>
<td align="right">0</td>
<td align="right">2.0e-07</td>
</tr>
<tr class="even">
<td align="left">353187</td>
<td align="right">0.6825313</td>
<td align="right">-0.3539853</td>
<td align="right">9.422013</td>
<td align="right">63.33253</td>
<td align="right">0</td>
<td align="right">2.0e-07</td>
</tr>
<tr class="odd">
<td align="left">270166</td>
<td align="right">-1.0453062</td>
<td align="right">0.2676608</td>
<td align="right">12.112605</td>
<td align="right">57.38212</td>
<td align="right">0</td>
<td align="right">4.0e-07</td>
</tr>
<tr class="even">
<td align="left">207818</td>
<td align="right">-0.6412353</td>
<td align="right">-0.1518757</td>
<td align="right">10.482197</td>
<td align="right">57.05721</td>
<td align="right">0</td>
<td align="right">4.0e-07</td>
</tr>
<tr class="odd">
<td align="left">15486</td>
<td align="right">-0.7315828</td>
<td align="right">0.1282440</td>
<td align="right">12.503645</td>
<td align="right">44.84398</td>
<td align="right">0</td>
<td align="right">3.5e-06</td>
</tr>
</tbody>
</table>
</div>
<div id="identify-differentially-expressed-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-differentially-expressed-genes" class="anchor"></a>Identify differentially expressed genes</h2>
<p>Differential expression is based on the coefficient for <code>cond</code> in a linear model with no interaction terms. We pass all genes to limma, but keep results only for non-differentially rhythmic genes, and adjust for multiple testing accordingly.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">design</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span>(~ <span class="no">cond</span> + <span class="no">time_cos</span> + <span class="no">time_sin</span>, <span class="kw">data</span> <span class="kw">=</span> <span class="no">sm</span>)

<span class="no">fit</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span>(<span class="no">emat</span>, <span class="no">design</span>)
<span class="no">fit</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span>(<span class="no">fit</span>, <span class="kw">trend</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="no">deLimma</span> <span class="kw">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html">data.table</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span>(<span class="no">fit</span>, <span class="kw">coef</span> <span class="kw">=</span> <span class="fl">2</span>, <span class="kw">number</span> <span class="kw">=</span> <span class="fl">Inf</span>), <span class="kw">keep.rownames</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setattr.html">setnames</a></span>(<span class="no">deLimma</span>, <span class="st">'rn'</span>, <span class="st">'geneId'</span>)

<span class="no">deLimma</span> <span class="kw">=</span> <span class="no">deLimma</span>[!(<span class="no">geneId</span> <span class="kw">%in%</span> <span class="no">drLimma</span>[<span class="no">adj.P.Val</span> <span class="kw">&lt;=</span> <span class="no">qvalDrCutoff</span>]$<span class="no">geneId</span>)]
<span class="no">deLimma</span>[, <span class="no">adj.P.Val</span> <span class="kw">:=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span>(<span class="no">P.Value</span>, <span class="kw">method</span> <span class="kw">=</span> <span class="st">'BH'</span>)]
<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setorder.html">setorderv</a></span>(<span class="no">deLimma</span>, <span class="st">'adj.P.Val'</span>)

<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">deLimma</span>[<span class="fl">1</span>:<span class="fl">5</span>, ])</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left">geneId</th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
<th align="right">B</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">11302</td>
<td align="right">-2.271205</td>
<td align="right">11.260315</td>
<td align="right">-23.07457</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">33.11920</td>
</tr>
<tr class="even">
<td align="left">68680</td>
<td align="right">-1.653067</td>
<td align="right">9.096749</td>
<td align="right">-18.20841</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">27.88414</td>
</tr>
<tr class="odd">
<td align="left">67442</td>
<td align="right">-1.208461</td>
<td align="right">14.375311</td>
<td align="right">-16.67155</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">25.88570</td>
</tr>
<tr class="even">
<td align="left">71904</td>
<td align="right">-1.457833</td>
<td align="right">10.350755</td>
<td align="right">-16.00387</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">24.95564</td>
</tr>
<tr class="odd">
<td align="left">15507</td>
<td align="right">-1.158876</td>
<td align="right">9.441377</td>
<td align="right">-15.27874</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">23.89967</td>
</tr>
</tbody>
</table>
</div>
<div id="plot-the-results" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-results" class="anchor"></a>Plot the results</h2>
<p>Numerous plots are possible. One is a volcano plot of differentially expressed genes.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">deLimma</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">logFC</span>, <span class="kw">y</span> <span class="kw">=</span> -<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">adj.P.Val</span>)), <span class="kw">size</span> <span class="kw">=</span> <span class="fl">0.2</span>, <span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">0.5</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span>(<span class="no">log</span>[<span class="fl">2</span>]*<span class="st">' fold-change'</span>), <span class="kw">y</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span>(-<span class="no">log</span>[<span class="fl">10</span>]*<span class="st">' '</span>*<span class="no">q</span>[<span class="no">DE</span>]))</pre></body></html></div>
<p><img src="introduction_files/figure-html/unnamed-chunk-11-1.png" width="384"></p>
<p>Another is a plot of expression as a function of time and genotype for individual genes. Here we show, from top to bottom, the top rhythmic gene, top differentially rhythmic gene, and top differentially expressed gene by q-value.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">geneIdsNow</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">rhyLimmaSummary</span>$<span class="no">geneId</span>[<span class="fl">1</span>], <span class="no">drLimma</span>$<span class="no">geneId</span>[<span class="fl">1</span>], <span class="no">deLimma</span>$<span class="no">geneId</span>[<span class="fl">1</span>])
<span class="no">geneSymbolsNow</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html">unname</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/annotate/man/getSYMBOL.html">getSYMBOL</a></span>(<span class="no">geneIdsNow</span>, <span class="st">'org.Mm.eg.db'</span>))

<span class="no">df</span> <span class="kw">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html">data.table</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="no">emat</span>[<span class="no">geneIdsNow</span>, ]))
<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/setattr.html">setnames</a></span>(<span class="no">df</span>, <span class="no">geneSymbolsNow</span>)
<span class="no">df</span>[, <span class="no">sample</span> <span class="kw">:=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">emat</span>[<span class="no">geneIdsNow</span>, ])]

<span class="no">df</span> <span class="kw">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/merge.html">merge</a></span>(<span class="no">df</span>, <span class="no">sm</span>[, <span class="fu">.</span>(<span class="no">sample</span>, <span class="no">cond</span>, <span class="no">time</span>)], <span class="kw">by</span> <span class="kw">=</span> <span class="st">'sample'</span>)
<span class="no">df</span> <span class="kw">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/melt.data.table.html">melt</a></span>(<span class="no">df</span>, <span class="kw">measure.vars</span> <span class="kw">=</span> <span class="no">geneSymbolsNow</span>, <span class="kw">variable.name</span> <span class="kw">=</span> <span class="st">'geneSymbol'</span>,
          <span class="kw">value.name</span> <span class="kw">=</span> <span class="st">'expr'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span>(<span class="no">df</span>$<span class="no">geneSymbol</span>) <span class="kw">=</span> <span class="no">geneSymbolsNow</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">df</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span>(<span class="no">geneSymbol</span> ~ <span class="no">cond</span>, <span class="kw">scales</span> <span class="kw">=</span> <span class="st">'free_y'</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">time</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">expr</span>, <span class="kw">shape</span> <span class="kw">=</span> <span class="no">cond</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="no">geneSymbol</span>), <span class="kw">size</span> <span class="kw">=</span> <span class="fl">2</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="st">'Zeitgeber time (h)'</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="st">'Expression (norm.)'</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_shape.html">scale_shape</a></span>(<span class="kw">solid</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">guide</span> <span class="kw">=</span> <span class="fl">FALSE</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_color_brewer</a></span>(<span class="kw">type</span> <span class="kw">=</span> <span class="st">'qual'</span>, <span class="kw">palette</span> <span class="kw">=</span> <span class="st">'Dark2'</span>, <span class="kw">guide</span> <span class="kw">=</span> <span class="fl">FALSE</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_continuous</a></span>(<span class="kw">limits</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">24</span>), <span class="kw">breaks</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>, <span class="fl">24</span>, <span class="fl">4</span>))</pre></body></html></div>
<p><img src="introduction_files/figure-html/unnamed-chunk-12-1.png" width="480"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jake Hughey.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
